aiofreepybox/aiofreepybox.py

```diff
@@ -63,6 +63,48 @@ def __init__(self, app_desc=app_desc, token_file=token_file, api_version='v3', t
         conn = aiohttp.TCPConnector(ssl_context=ssl_ctx)
         self._session = aiohttp.ClientSession(connector=conn)
 
+        # Detect host, port and api_version
+        detect = [
+            host,
+            port,
+            self.api_version
+            ]
+        if 'auto' in detect:
+            default_host = '212.27.38.253'
+            default_host_list = [
+                'auto',
+                'mafreebox.freebox.fr',
+                '192.168.0.254',
+                default_host
+                ]
+            if host not in default_host_list:
+                default_host = host
+                logger.debug(f'Host set to {host}')
+                if port == 'auto':
+                    logger.warning('Port is set to auto, but host is not in default host list, checking port 80')
+                    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                    result = sock.connect_ex((default_host,80))
+                    if result != 0:
+                        raise HttpRequestError('Port 80 is closed, cannot detect freebox')
+            r = await self._session.get(f'http://{default_host}/api_version', timeout=self.timeout)
+            resp = await r.json()
+            if host == 'auto':
+                host = resp['api_domain']
+                logger.debug(f'Host set to {host}')
+            if port == 'auto':
+                port = resp['https_port']
+                logger.debug(f'Port set to {port}')
+            server_version = resp['api_version'][:1]
```

Comment:
Maybe split on `.` and get the first token rather than expecting the version to fit on a single digit (I assume version 10.0 will happen sooner or later :) )

Reply:
It could happen around 2023, i'll split it.